private/bufpkg/bufpluginexec/bufpluginexec.go

```diff
@@ -206,3 +218,9 @@ type handlerOptions struct {
 func newHandlerOptions() *handlerOptions {
 	return &handlerOptions{}
 }
+
+// looksLikeWASM is a minimal check for WASM plugins. A more stringent validation
+// of the file is done in the handlers Handle method.
+func looksLikeWASM(pluginName string) bool {
```

Comment:
This feels a little ish imo - we should have a definitive test before we do any logic here

Reply:
I am adding comments around this check.